packages/core/core-flows/src/payment/workflows/refund-payment.ts

```diff
@@ -11,6 +12,41 @@ import { emitEventStep, useRemoteQueryStep } from "../../common"
 import { addOrderTransactionStep } from "../../order/steps/add-order-transaction"
 import { refundPaymentStep } from "../steps/refund-payment"
 
+/**
+ * This step validates that the refund is valid for the order
+ */
+export const validateRefundStep = createStep(
+  "begin-claim-order-validation",
+  async function ({
+    order,
+    payment,
+    amount,
+  }: {
+    order: OrderDTO
+    payment: PaymentDTO
+    amount?: BigNumberInput
+  }) {
+    const pendingDifference = order.summary?.raw_pending_difference!
+
+    if (MathBN.gte(pendingDifference, 0)) {
```

Comment:
question: Can the pending different be greater than in this scenario? Wouldn't it always be 0 or am I reading this wrong?

Reply:
yes, it can be a positive number as well in case there is pending payment from the customer side

Reply:
Got it!